app/views/users/show.html.erb

```diff
@@ -24,15 +24,35 @@
       <section class="followers">
         <h2>Followers</h2>
         <% @github_user.followers.each do |follower| %>
-          <%= link_to follower.handle, follower.url %><br>
+          <%= link_to follower.handle, follower.url %>
+          <% if follower.in_system? && !follower.already_friends?(current_user.id) %>
+            <%= link_to "Add as Friend", "users/#{current_user.id}/friendships?future_friend=#{follower.handle}", method: :post, class: "btn btn-primary mb1 bg-teal" %><br>
+          <% else %><br>
+          <% end %>
         <% end %>
       </section>
       <section class="following">
         <h2>Following</h2>
         <% @github_user.followings.each do |following| %>
           <%= link_to following.handle, following.url %><br>
+          <% if following.in_system? && !following.already_friends?(current_user.id) %>
+            <%= link_to "Add as Friend", "users/#{current_user.id}/friendships?future_friend=#{following.handle}", method: :post, class: "btn btn-primary mb1 bg-teal" %><br>
+          <% else %><br>
+          <% end %>
         <% end %>
       </section>
-    <%end %>
+    <% end %>
+  </section>
+
+  <section class="friendships">
+    <h1>Friends</h1>
+    <% if current_user.friends.empty? %>
+      <%= "Bummer.You have no friends." %>
```

Comment:
![](https://media1.giphy.com/media/niMQW6U668wcU/giphy.gif?cid=5a38a5a2bf7d6bc07f4215a776f2d6fd13ae37c4ff70859a&rid=giphy.gif)

Reply:
this will haunt me